remix-analyzer/src/solidity-analyzer/modules/selfdestruct.js

```diff
@@ -2,25 +2,51 @@ var name = 'Selfdestruct: '
 var desc = 'Be aware of caller contracts.'
 var categories = require('./categories')
 var common = require('./staticAnalysisCommon')
+var AbstractAst = require('./abstractAstView')
 
 function selfdestruct () {
-  this.relevantNodes = []
-}
+  this.abstractAst = new AbstractAst()
+
+  this.visit = this.abstractAst.build_visit(
+    (node) => common.isStatement(node) ||
+              common.isSelfdestructCall(node)
+  )
 
-selfdestruct.prototype.visit = function (node) {
-  if (common.isSelfdestructCall(node)) {
-    this.relevantNodes.push(node)
-  }
+  this.report = this.abstractAst.build_report(report)
 }
 
-selfdestruct.prototype.report = function () {
-  return this.relevantNodes.map(function (item, i) {
-    return {
-      warning: 'Use of selfdestruct: can block calling contracts unexpectedly. Be especially careful if this contract is planned to be used by other contracts (i.e. library contracts, interactions). Selfdestruction of the callee contract can leave callers in an inoperable state.',
-      location: item.src,
-      more: 'https://paritytech.io/blog/security-alert.html'
-    }
+selfdestruct.prototype.visit = function () { throw new Error('constantFunctions.js no visit function set upon construction') }
+
+selfdestruct.prototype.report = function () { throw new Error('constantFunctions.js no report function set upon construction') }
+
+function report (contracts, multipleContractsWithSameName) {
+  var warnings = []
+
+  contracts.forEach((contract) => {
+    contract.functions.forEach((func) => {
+      let hasSelf = false
+      func.relevantNodes.forEach((node) => {
+        if (common.isSelfdestructCall(node)) {
+          warnings.push({
+            warning: 'Use of selfdestruct: can block calling contracts unexpectedly. Be especially careful if this contract is planned to be used by other contracts (i.e. library contracts, interactions). Selfdestruction of the callee contract can leave callers in an inoperable state.',
+            location: node.src,
+            more: 'https://paritytech.io/blog/security-alert.html'
+          })
+          hasSelf = true
+        }
+        if (common.isStatement(node) && hasSelf) {
+          warnings.push({
```

Comment:
Would it make sense to add a `false positive possible` flag in these kind of report or module? 
Some other module also give false positive (the dao check afar?), and some users think there are real issues and try to them it even though that might not apply to their code.

Reply:
I totally agree with you. I will add some flags to the module description. They can then be used by the UI.